lib/types/request.d.ts

```diff
@@ -27,23 +27,24 @@ export interface AppCredentials {
  * User-extensible type for request.auth credentials.
  */
 export interface AuthCredentials<
-    AuthUser extends object = UserCredentials,
```

Comment:
Technically you can say that User is a string and hapi will accept it

Reply:
It looks like the types were trying to be prescriptive in "good" practices, should we just stick with what hapi does then?

Reply:
The reason I removed it was because, to me, it's like the decorator types where it expects you to pass a function but in reality you can pass anything and it will decorate. I think we should type reality, no? IFAIK, technically, `credentials` and `artifacts` can be a single primitive value; so can `app`, so can `user`. Hapi does not internally use any of these values. These are user-defined and I think the user should dictate how they need to build their auth.

Example: 

https://github.com/hapijs/basic/blob/master/lib/index.js#L64
https://github.com/hapijs/jwt/blob/master/lib/plugin.js#L225

They're never used and can be anything at runtime.


However, if you think we should communicate that these must be objects, I won't disagree either because it's not wrong. I've never not-used an object in these abstractions.